src/components/ConnectWallet/ConnectWallet.tsx

```diff
@@ -102,19 +124,24 @@ const ConnectWallet = () => {
     dispatch(logout())
   }
 
-  const initiatePrivateWalletSignature = async (currentWallet: any, walletAddr_bech32: any, walletAddr: string) => {
+  const initiatePrivateWalletSignature = async (
+    currentWallet: any,
+    stakeAddressBech32: StakeAddressBech32,
+    stakeAddressHex: StakeAddressHex,
+    changeAddressBech32: ChangeAddressBech32
+  ) => {
     const timestamp = (await fetchData.get<any,AxiosResponse<number>,any>('/server-timestamp')).data;
-    const msgToBeSigned = `Sign this message if you are the owner of the ${walletAddr_bech32} address. \n Timestamp: <<${timestamp}>> \n Expiry: 60 seconds`;
+    const msgToBeSigned = `Sign this message if you are the owner of the ${changeAddressBech32} address. \n Timestamp: <<${timestamp}>> \n Expiry: 60 seconds`;
     try {
-        const {key, signature} = await currentWallet.signData(walletAddr, Buffer.from(msgToBeSigned, 'utf8').toString('hex'))
+        const {key, signature} = await currentWallet.signData(stakeAddressHex, Buffer.from(msgToBeSigned, 'utf8').toString('hex'))
         if (key && signature) {
             const token = (await fetchData.post<any,AxiosResponse<string>,any>('/login', {
-                address: walletAddr_bech32,
+                address: stakeAddressBech32,
                 key: key,
                 signature: signature
             })).data;
             localStorage.setItem(LocalStorageKeys.authToken, token)
-            setAddress(walletAddr_bech32)
+            setAddress(changeAddressBech32)
```

Comment:
Since we are storing the `stakeAddress` in BE as unique identify of the wallet, wouldn't it be better to store this in FE as well; instead of `changeAddress` ?

Reply:
we still need the address to make payments